--- a/net/minecraft/server/ReloadableServerResources.java
+++ b/net/minecraft/server/ReloadableServerResources.java
@@ -77,17 +77,23 @@
         Executor p_249136_,
         Executor p_249601_
     ) {
+        System.out.println("[mc-web/reload] loadResources begin");
         return ReloadableServerRegistries.reload(p_330376_, p_366334_, p_248588_, p_249136_)
             .thenCompose(
                 p_358539_ -> {
+                    System.out.println("[mc-web/reload] registries reload end");
                     ReloadableServerResources reloadableserverresources = new ReloadableServerResources(
                         p_358539_.layers(), p_358539_.lookupWithUpdatedTags(), p_250212_, p_249301_, p_366334_, p_251126_
                     );
+                    System.out.println("[mc-web/reload] listeners create begin");
                     return SimpleReloadInstance.create(
                             p_248588_, reloadableserverresources.listeners(), p_249136_, p_249601_, DATA_RELOAD_INITIAL_TASK, LOGGER.isDebugEnabled()
                         )
                         .done()
-                        .thenApply(p_214306_ -> reloadableserverresources);
+                        .thenApply(p_214306_ -> {
+                            System.out.println("[mc-web/reload] listeners done");
+                            return reloadableserverresources;
+                        });
                 }
             );
     }
@@ -95,4 +101,4 @@
     public void updateStaticRegistryTags() {
         this.postponedTags.forEach(Registry.PendingTags::apply);
     }
-}
\ No newline at end of file
+}
