--- a/net/minecraft/data/worldgen/placement/PlacementUtils.java
+++ b/net/minecraft/data/worldgen/placement/PlacementUtils.java
@@ -25,6 +25,7 @@
 import net.minecraft.world.level.levelgen.placement.PlacedFeature;
 import net.minecraft.world.level.levelgen.placement.PlacementFilter;
 import net.minecraft.world.level.levelgen.placement.PlacementModifier;
+import top.steve3184.webmc.web.BuildFlags;
 
 public class PlacementUtils {
     public static final PlacementModifier HEIGHTMAP = HeightmapPlacement.onHeightmap(Heightmap.Types.MOTION_BLOCKING);
@@ -74,11 +75,13 @@
 
     public static PlacementModifier countExtra(int p_195365_, float p_195366_, int p_195367_) {
         float f = 1.0F / p_195366_;
-        if (Math.abs(f - (int)f) > 1.0E-5F) {
+        int i = Math.round(f);
+        float f1 = Math.abs(f - (float)i);
+        if (f1 > (BuildFlags.WEB_RUNTIME ? 1.0E-3F : 1.0E-5F)) {
             throw new IllegalStateException("Chance data cannot be represented as list weight");
         } else {
             WeightedList<IntProvider> weightedlist = WeightedList.<IntProvider>builder()
-                .add(ConstantInt.of(p_195365_), (int)f - 1)
+                .add(ConstantInt.of(p_195365_), i - 1)
                 .add(ConstantInt.of(p_195365_ + p_195367_), 1)
                 .build();
             return CountPlacement.of(new WeightedListInt(weightedlist));
@@ -110,4 +113,4 @@
     public static <FC extends FeatureConfiguration, F extends Feature<FC>> Holder<PlacedFeature> filtered(F p_206499_, FC p_206500_, BlockPredicate p_206501_) {
         return inlinePlaced(p_206499_, p_206500_, BlockPredicateFilter.forPredicate(p_206501_));
     }
-}
\ No newline at end of file
+}
