--- a/org\lwjgl\openal\ALCCapabilities.java
+++ b/org\lwjgl\openal\ALCCapabilities.java
@@ -0,0 +1,11 @@
+package org.lwjgl.openal;
+
+public final class ALCCapabilities {
+    public final boolean OpenALC10 = true;
+    public final boolean OpenALC11 = true;
+    public final boolean ALC_EXT_disconnect = false;
+    public final boolean ALC_EXT_thread_local_context = false;
+    public final boolean ALC_SOFT_HRTF = false;
+    public final boolean ALC_SOFT_pause_device = false;
+    public final boolean ALC_SOFT_output_limiter = false;
+}
