--- a/net/minecraft/client/quickplay/QuickPlayLog.java
+++ b/net/minecraft/client/quickplay/QuickPlayLog.java
@@ -42,7 +42,7 @@
     private QuickPlayLog.QuickPlayWorld worldData;
 
     QuickPlayLog(String p_279463_) {
-        this.path = Minecraft.getInstance().gameDirectory.toPath().resolve(p_279463_);
+        this.path = java.nio.file.Paths.get(Minecraft.getInstance().gameDirectory.getPath()).resolve(p_279463_);
     }
 
     public static QuickPlayLog of(@Nullable String p_279275_) {
