--- a/net/minecraft/client/resources/SkinManager.java
+++ b/net/minecraft/client/resources/SkinManager.java
@@ -29,6 +29,7 @@
 import net.minecraftforge.api.distmarker.Dist;
 import net.minecraftforge.api.distmarker.OnlyIn;
 import org.slf4j.Logger;
+import top.steve3184.webmc.web.BuildFlags;
 
 @OnlyIn(Dist.CLIENT)
 public class SkinManager {
@@ -44,6 +45,10 @@
         this.skinTextures = new SkinManager.TextureCache(p_299617_, Type.SKIN);
         this.capeTextures = new SkinManager.TextureCache(p_299617_, Type.CAPE);
         this.elytraTextures = new SkinManager.TextureCache(p_299617_, Type.ELYTRA);
+        if (BuildFlags.WEB_RUNTIME) {
+            this.skinCache = null;
+            return;
+        }
         this.skinCache = CacheBuilder.newBuilder()
             .expireAfterAccess(Duration.ofSeconds(15L))
             .build(
@@ -93,6 +98,9 @@
     }
 
     public CompletableFuture<Optional<PlayerSkin>> getOrLoad(GameProfile p_298661_) {
+        if (BuildFlags.WEB_RUNTIME) {
+            return CompletableFuture.completedFuture(Optional.empty());
+        }
         Property property = this.sessionService.getPackedTextures(p_298661_);
         return this.skinCache.getUnchecked(new SkinManager.CacheKey(p_298661_.getId(), property));
     }
@@ -174,4 +182,4 @@
             return ResourceLocation.withDefaultNamespace(s + "/" + p_297392_);
         }
     }
-}
\ No newline at end of file
+}
