--- a/net/minecraft/client/gui/screens/CreateBuffetWorldScreen.java
+++ b/net/minecraft/client/gui/screens/CreateBuffetWorldScreen.java
@@ -1,8 +1,6 @@
 package net.minecraft.client.gui.screens;
 
-import com.ibm.icu.text.Collator;
 import java.util.Comparator;
-import java.util.Locale;
 import java.util.Objects;
 import java.util.function.Consumer;
 import javax.annotation.Nullable;
@@ -84,7 +82,7 @@
     class BiomeList extends ObjectSelectionList<CreateBuffetWorldScreen.BiomeList.Entry> {
         BiomeList() {
             super(CreateBuffetWorldScreen.this.minecraft, CreateBuffetWorldScreen.this.width, CreateBuffetWorldScreen.this.height - 77, 40, 16);
-            Collator collator = Collator.getInstance(Locale.getDefault());
+            Comparator<String> collator = String.CASE_INSENSITIVE_ORDER;
             CreateBuffetWorldScreen.this.biomes
                 .listElements()
                 .map(p_205389_ -> new CreateBuffetWorldScreen.BiomeList.Entry((Holder.Reference<Biome>)p_205389_))
