diff --git a/net/minecraft/client/gui/screens/ReceivingLevelScreen.java b/net/minecraft/client/gui/screens/ReceivingLevelScreen.java
index 0e9ef68..d2bb2a9 100644
--- a/net/minecraft/client/gui/screens/ReceivingLevelScreen.java
+++ b/net/minecraft/client/gui/screens/ReceivingLevelScreen.java
@@ -15,6 +15,7 @@ import net.minecraft.network.chat.Component;
 import net.minecraft.world.level.block.Blocks;
 import net.minecraftforge.api.distmarker.Dist;
 import net.minecraftforge.api.distmarker.OnlyIn;
+import top.steve3184.webmc.web.BuildFlags;
 
 @OnlyIn(Dist.CLIENT)
 public class ReceivingLevelScreen extends Screen {
@@ -51,6 +52,9 @@ public class ReceivingLevelScreen extends Screen {
 
     @Override
     public void renderBackground(GuiGraphics p_298240_, int p_297552_, int p_298125_, float p_297335_) {
+        if (BuildFlags.WEB_RUNTIME) {
+            return;
+        }
         switch (this.reason) {
             case NETHER_PORTAL:
                 p_298240_.blitSprite(RenderPipelines.GUI_OPAQUE_TEXTURED_BACKGROUND, this.getNetherPortalSprite(), 0, 0, p_298240_.guiWidth(), p_298240_.guiHeight());
@@ -80,14 +84,16 @@ public class ReceivingLevelScreen extends Screen {
 
     @Override
     public void tick() {
-        if (this.levelReceived.getAsBoolean() || Util.getMillis() > this.createdAt + 30000L) {
+        if (this.levelReceived.getAsBoolean() || (!BuildFlags.WEB_RUNTIME && Util.getMillis() > this.createdAt + CHUNK_LOADING_START_WAIT_LIMIT_MS)) {
             this.onClose();
         }
     }
 
     @Override
     public void onClose() {
-        this.minecraft.getNarrator().saySystemNow(Component.translatable("narrator.ready_to_play"));
+        if (!BuildFlags.WEB_RUNTIME) {
+            this.minecraft.getNarrator().saySystemNow(Component.translatable("narrator.ready_to_play"));
+        }
         super.onClose();
     }
 
@@ -102,4 +108,4 @@ public class ReceivingLevelScreen extends Screen {
         END_PORTAL,
         OTHER;
     }
-}
\ No newline at end of file
+}
