--- a/net/minecraft/client/telemetry/ClientTelemetryManager.java
+++ b/net/minecraft/client/telemetry/ClientTelemetryManager.java
@@ -50,7 +50,7 @@
         telemetrypropertymap$builder.put(TelemetryProperty.CLIENT_MODDED, Minecraft.checkModStatus().shouldReportAsModified());
         telemetrypropertymap$builder.putIfNotNull(TelemetryProperty.LAUNCHER_NAME, Minecraft.getLauncherBrand());
         this.deviceSessionProperties = telemetrypropertymap$builder.build();
-        this.logDirectory = p_261610_.gameDirectory.toPath().resolve("logs/telemetry");
+        this.logDirectory = java.nio.file.Paths.get(p_261610_.gameDirectory.getPath()).resolve("logs/telemetry");
         this.logManager = TelemetryLogManager.open(this.logDirectory);
     }
 
