--- a/net/minecraft/network/chat/PlayerChatMessage.java
+++ b/net/minecraft/network/chat/PlayerChatMessage.java
@@ -34,7 +34,7 @@
                 )
             )
     );
-    private static final UUID SYSTEM_SENDER = Util.NIL_UUID;
+    private static final UUID SYSTEM_SENDER = new UUID(0L, 0L);
     public static final Duration MESSAGE_EXPIRES_AFTER_SERVER = Duration.ofMinutes(5L);
     public static final Duration MESSAGE_EXPIRES_AFTER_CLIENT = MESSAGE_EXPIRES_AFTER_SERVER.plus(Duration.ofMinutes(2L));
 
