--- a/org\lwjgl\glfw\GLFWCursorPosCallback.java
+++ b/org\lwjgl\glfw\GLFWCursorPosCallback.java
@@ -0,0 +1,6 @@
+package org.lwjgl.glfw;
+public abstract class GLFWCursorPosCallback implements GLFWCursorPosCallbackI {
+    public static GLFWCursorPosCallback create(GLFWCursorPosCallbackI l) { return new GLFWCursorPosCallback() { @Override public void invoke(long w, double x, double y) { l.invoke(w, x, y); } }; }
+    public void free() {}
+    public GLFWCursorPosCallback set(long window) { GLFW.glfwSetCursorPosCallback(window, this); return this; }
+}
