--- a/net/minecraft/network/protocol/game/ClientboundLevelChunkPacketData.java
+++ b/net/minecraft/network/protocol/game/ClientboundLevelChunkPacketData.java
@@ -32,6 +32,8 @@ public class ClientboundLevelChunkPacketData {
     private final Map<Heightmap.Types, long[]> heightmaps;
     private final byte[] buffer;
     private final List<ClientboundLevelChunkPacketData.BlockEntityInfo> blockEntitiesData;
+    @Nullable
+    private final LevelChunk webChunk;
 
     public ClientboundLevelChunkPacketData(LevelChunk p_195651_) {
         this.heightmaps = p_195651_.getHeightmaps()
@@ -45,6 +47,7 @@ public class ClientboundLevelChunkPacketData {
         for (Entry<BlockPos, BlockEntity> entry : p_195651_.getBlockEntities().entrySet()) {
             this.blockEntitiesData.add(ClientboundLevelChunkPacketData.BlockEntityInfo.create(entry.getValue()));
         }
+        this.webChunk = p_195651_;
     }
 
     public ClientboundLevelChunkPacketData(RegistryFriendlyByteBuf p_335775_, int p_195654_, int p_195655_) {
@@ -56,6 +59,7 @@ public class ClientboundLevelChunkPacketData {
             this.buffer = new byte[i];
             p_335775_.readBytes(this.buffer);
             this.blockEntitiesData = ClientboundLevelChunkPacketData.BlockEntityInfo.LIST_STREAM_CODEC.decode(p_335775_);
+            this.webChunk = null;
         }
     }
 
@@ -118,6 +122,11 @@ public class ClientboundLevelChunkPacketData {
     public Map<Heightmap.Types, long[]> getHeightmaps() {
         return this.heightmaps;
     }
+
+    @Nullable
+    public LevelChunk getWebChunk() {
+        return this.webChunk;
+    }
 
     static class BlockEntityInfo {
         public static final StreamCodec<RegistryFriendlyByteBuf, ClientboundLevelChunkPacketData.BlockEntityInfo> STREAM_CODEC = StreamCodec.ofMember(
