--- a/net/minecraft/client/gui/components/toasts/SystemToast.java
+++ b/net/minecraft/client/gui/components/toasts/SystemToast.java
@@ -115,10 +115,16 @@
     }
 
     public static void add(ToastManager p_362779_, SystemToast.SystemToastId p_94857_, Component p_94858_, @Nullable Component p_94859_) {
+        if (p_362779_ == null) {
+            return;
+        }
         p_362779_.addToast(new SystemToast(p_94857_, p_94858_, p_94859_));
     }
 
     public static void addOrUpdate(ToastManager p_360727_, SystemToast.SystemToastId p_94871_, Component p_94872_, @Nullable Component p_94873_) {
+        if (p_360727_ == null) {
+            return;
+        }
         SystemToast systemtoast = p_360727_.getToast(SystemToast.class, p_94871_);
         if (systemtoast == null) {
             add(p_360727_, p_94871_, p_94872_, p_94873_);
@@ -128,6 +134,9 @@
     }
 
     public static void forceHide(ToastManager p_366670_, SystemToast.SystemToastId p_311637_) {
+        if (p_366670_ == null) {
+            return;
+        }
         SystemToast systemtoast = p_366670_.getToast(SystemToast.class, p_311637_);
         if (systemtoast != null) {
             systemtoast.forceHide();
@@ -205,4 +214,4 @@
             this(5000L);
         }
     }
-}
\ No newline at end of file
+}
