--- a/com/mojang/blaze3d/platform/GLX.java
+++ b/com/mojang/blaze3d/platform/GLX.java
@@ -20,8 +20,6 @@
 import org.lwjgl.glfw.GLFWVidMode;
 import org.lwjgl.system.MemoryUtil;
 import org.slf4j.Logger;
-import oshi.SystemInfo;
-import oshi.hardware.CentralProcessor;
 
 @OnlyIn(Dist.CLIENT)
 @DontObfuscate
@@ -81,14 +79,8 @@
 
     public static String _getCpuInfo() {
         if (cpuInfo == null) {
-            cpuInfo = "<unknown>";
-
-            try {
-                CentralProcessor centralprocessor = new SystemInfo().getHardware().getProcessor();
-                cpuInfo = String.format(Locale.ROOT, "%dx %s", centralprocessor.getLogicalProcessorCount(), centralprocessor.getProcessorIdentifier().getName())
-                    .replaceAll("\\s+", " ");
-            } catch (Throwable throwable) {
-            }
+            cpuInfo = String.format(Locale.ROOT, "%dx %s", Runtime.getRuntime().availableProcessors(), System.getProperty("os.arch", "unknown"))
+                .replaceAll("\\s+", " ");
         }
 
         return cpuInfo;
@@ -102,4 +94,4 @@
         p_69372_.accept(p_69371_);
         return p_69371_;
     }
-}
\ No newline at end of file
+}
