diff --git a/net/minecraft/server/level/ChunkTrackingView.java b/net/minecraft/server/level/ChunkTrackingView.java
index 813b8c3..113e9c4 100644
--- a/net/minecraft/server/level/ChunkTrackingView.java
+++ b/net/minecraft/server/level/ChunkTrackingView.java
@@ -71,7 +71,10 @@ public interface ChunkTrackingView {
     }
 
     static boolean isWithinDistance(int p_299483_, int p_297415_, int p_300799_, int p_299157_, int p_301327_, boolean p_301271_) {
-        int i = p_301271_ ? 2 : 1;
+        int i = p_301271_
+            && !(top.steve3184.webmc.web.BuildFlags.WEB_RUNTIME && top.steve3184.webmc.web.BuildFlags.ENABLE_EXPERIMENTAL_MC_MAIN)
+                ? 2
+                : 1;
         long j = Math.max(0, Math.abs(p_299157_ - p_299483_) - i);
         long k = Math.max(0, Math.abs(p_301327_ - p_297415_) - i);
         long l = j * j + k * k;
@@ -120,4 +123,4 @@ public interface ChunkTrackingView {
             }
         }
     }
-}
\ No newline at end of file
+}
