--- a/net/minecraft/client/gui/Font.java
+++ b/net/minecraft/client/gui/Font.java
@@ -1,8 +1,5 @@
 package net.minecraft.client.gui;
 
-import com.ibm.icu.text.ArabicShaping;
-import com.ibm.icu.text.ArabicShapingException;
-import com.ibm.icu.text.Bidi;
 import com.mojang.blaze3d.font.GlyphInfo;
 import com.mojang.blaze3d.vertex.VertexConsumer;
 import java.util.ArrayList;
@@ -57,13 +54,7 @@
     }
 
     public String bidirectionalShaping(String p_92802_) {
-        try {
-            Bidi bidi = new Bidi(new ArabicShaping(8).shape(p_92802_), 127);
-            bidi.setReorderingMode(0);
-            return bidi.writeReordered(2);
-        } catch (ArabicShapingException arabicshapingexception) {
-            return p_92802_;
-        }
+        return p_92802_;
     }
 
     public void drawInBatch(
